### ISSUE TITLE
Depending on boot.mount by other initramfs-stage units (mounts) fails because dracut mount /boot before systemd (error code 32, already mounted on /boot)

### ISSUE THREAD

**Issue by ghost** [systemd/systemd#8793]
### Submission type

 - Bug report

### systemd version the issue has been seen with

238

I forgot which exact release version my installation uses, because it's broken for now and not booting. And I'm using dual-booted gaming-Windows to report this. I can dig it out if you need.

### Used distribution

Fedora Atomic Workstation 28 (yes, I know, kind of special, but the issue is not with ostree)

### Expected behaviour you didn't see

If `/boot` partition is already mounted on that mount point earlier by something, just ignore the error code, log the warning about that fact and mark unit as completed/success. Or add an option to enable such behavior for that particular unit, like `IgnoreAlreadyMounted`.

### Unexpected behaviour you saw

Because dracut mounted `/boot` partition earlier than systemd was launched and `boot.mount` unit started, unit enters failed state and everything that depends on it also chained. And in log it gives error message "devid is already mounted on /boot", which is looking silly.

### Steps to reproduce the problem

1. Add `x-systemd.requires=boot.mount,x-systemd.after=boot.mount` or `x-systemd.requires-mounts-for=/boot` to options of any fstab record, except root filesystem. To `/boot/efi` record, for example. Or to the any of the crypttab records, because I placed the gpg encrypted key to my LUKS setup on `/boot`.
1. Force dracut to mount `/boot`, maybe, by adding a crypttab entry with key located on `/boot`. Because I don't know, is this the default behavior or it's me forced to mount `/boot` by placing key there.
1. Force rebuild the dracut image and reboot the system.
1. ???
1. Everything is broken, because `/boot` is already mounted.

P.S. I know the fact that `/boot/efi` is kind of logically includes mounting `/boot` fisrt. But I like to be "Captain Obvious" sometimes, even in fstab. And it's good for safety, because systemd mount everything in parallel.

**Comment by poettering** [comment]
systemd won't mount stuff again that is already mounted, it watches /proc/self/mountinfo for that, hence there must be another explanation for what you are seeing.

What are the precise error messages you are seeing?

**Comment by poettering** [comment]
And to say this clearly: we support that arbitrary stuff is premounted when the system transitions from the initrd into main host file system. It is in fact how we recommend compatibility with systems that have /usr split off and suchlike

**Comment by nexfwall** [comment]
Sorry for delay in responding.

> systemd won't mount stuff again that is already mounted, it watches /proc/self/mountinfo for that, hence there must be another explanation for what you are seeing.
   What are the precise error messages you are seeing?

I think rdosreport will tell better than me.
[rdsosreport.txt](https://github.com/systemd/systemd/files/1956928/rdsosreport.txt)

It's old, because I wasn't trying booting the system since than. Wanted to reach out to you or your team first. But was busy with usual stuff.

Just search for `boot.mount` and you will see everything. Maybe I was blaming wrong one from the start?

```
[    9.281511] localhost systemd[1]: boot.mount: Directory /boot to mount over is not empty, mounting anyway.
[    9.281745] localhost systemd-fsck[1162]: CP437: Invalid argument
[    9.281983] localhost systemd[1]: Mounting /boot...
[    9.283595] localhost mount[1179]: mount: /boot: /dev/nvme0n1p4 already mounted on /boot.
[    9.283826] localhost systemd[1]: boot.mount: Mount process exited, code=exited status=32
[    9.283863] localhost systemd[1]: boot.mount: Failed with result 'exit-code'.
[    9.284014] localhost systemd[1]: Failed to mount /boot.
```

**Comment by nexfwall** [comment]
Well, I think in this situation my configs might be very useful.

[crypttab.txt](https://github.com/systemd/systemd/files/1960741/crypttab.txt)
[fstab.txt](https://github.com/systemd/systemd/files/1960742/fstab.txt)
##### `dracut.conf.d`

* [90-always_use_fstab.conf.txt](https://github.com/systemd/systemd/files/1960746/90-always_use_fstab.conf.txt)
* [91-add_crypt-gpg.conf.txt](https://github.com/systemd/systemd/files/1960747/91-add_crypt-gpg.conf.txt)
* [92-add_swaptools.conf.txt](https://github.com/systemd/systemd/files/1960748/92-add_swaptools.conf.txt)

And I'll explain basically what I did:
I wanted to try to use *LUKS on LVM* setup, because encrypting everything and losing speed on such things like rootfs of an atomic system seemed ridiculous to me. My rootfs is located on very fast NVMe SSD. It's maximum speed is almost beyond of the capability of my processor to encrypt things. But rootfs contains nothing in need of protection. So...
Anaconda does not support such setups. And this is an Atomic system, it's not such a thing which can be manually installed by bare hands. So I've installed it as a normal system, with all partitions and filesystems I want, and begun converting it manually.
( Yes, I know I might be a masochist which likes to shoot it's own foot, but just deal with it. )

**Comment by nexfwall** [comment]
@poettering, is there some temporary (ugly) workaround possible? My system is still down, for weeks, and I can't wait forever for @haraldh to answer.

Also, if there is no any, I have another plan how to organize my system properly using "basic" LVM over LUKS and not so tricky as it is now. But this mean reinstalling the system again from scratch and loosing the only testing ground for solving this issue I have. And this means closing this issue without resolving.

I just can't sit on Windows forever. I need my Fedora back. 🙁
